compiler/src/dmd/dinterpret.d

```diff
@@ -3891,7 +3891,7 @@ public:
             newval = copyLiteral(newval).copy();
             assignInPlace(oldval, newval);
         }
-        else if (wantCopy && e.op == EXP.assign)
+        else if (wantCopy && (e.op == EXP.assign || e.op == EXP.loweredAssignExp))
```

Comment:
Maybe I am missing something here, but as far as I understand the interpreter should never have to deal with `LoweredAssignExp`.  That's why we are keeping the original expression, no?

Reply:
No, it should. In `expressionsem.d` the lowering changes the expression to `LoweredAssignExp` [1]. That's why the interpreter should handle it the same way it handles `AssignExp`s.

[1] https://github.com/dlang/dmd/blob/a528c995bf1c80acd4766002d9da198ab396989c/compiler/src/dmd/expressionsem.d#L10311-L10312